### DISCUSSION TITLE
Environment variables can not be accessed when deploying on

### DISCUSSION THREAD

**Discussion by ergunpp** [gatsbyjs/gatsby#27893] [Help]
I have a project that uses Gatsby default starter with fully dynamic app that fetches data through was api gateway and uses was cognito authentication.
Running locally and testing the build in Aws S3 everything works fine. 
But trying to deploy it through GitHub and Amplify, the environment variables referenced in the application are undefined.
I have entered the env variables both through the console and into the yaml file, but still the same problem.

<img width="926" alt="Screen Shot 2020-11-08 at 11 05 38" src="https://user-images.githubusercontent.com/24900951/98460127-84275e80-21b2-11eb-9d5a-62e8b84c96d8.png">
<img width="1736" alt="Screen Shot 2020-11-08 at 11 04 02" src="https://user-images.githubusercontent.com/24900951/98460129-87224f00-21b2-11eb-81d1-e3fd330672a7.png">

Referencing them in the code like so:

`const [isLoading, setIsLoading] = useState(true)
    const [orders, setOrders] = useState({})
    const path = process.env.ERP_API_URL
    const erp = process.env.ERP_COMP_NO
    const id = props.id
    const callback = props.callback
    const [error, setError] = useState(false)`

Need help. Thank you in advance

**Comment by wardpeet** [comment]
What do you get when you run build locally? I don't think the env vars are being passed to your application. Can you prefix them with GATSBY_ ?
